reflection.md

```diff
@@ -1,18 +1,28 @@
 # Reflection on implementing regular expressions of a DSL
 
 ## Which operators were easiest to implement and why?
+   The easiest operators to implement were the ones for Union, Concatenation. Once I implemented Union, doing Concatenation was trivial because it had the same structure. Kleene Star and <+> were a bit more interesting because I had to figure out how to define a postfix operator, in particular, how to access the value before the operator. The implementation on <+> was simply building on top of the Kleene Star to match one more literal/expression. 
 
 ## Which operators were most difficult to implement and why?
+   The hardest one to implement was the apply function to return a regular expression that matches n reptitions of the element. I wasn't aware that apply is a special function and was trying to write it with an implicit function. Implementing implicit functions within the companion object was also hard at first. While it wasn't that much actual code to implement, it was a little confusing initially to understand how it worked. 
```

Comment:
True. It was also tricky because we can omit the method name (`apply`), but we didn't know that at first